azure_gateway/_gateway_llm.py

```diff
@@ -0,0 +1,282 @@
+import json
+import logging
+import os
+import time
+from typing import Callable
+
+import requests
+import tiktoken
+from langchain_core.language_models.llms import LLM
+from openai import AzureOpenAI
+from requests_sse import EventSource
+
+from ._exceptions import OutOfQuotaError
+from ._logger import LoggerWrapper
+
+
+Callback = Callable[[str], None]
+ServiceResponse = tuple[str, int, int]
+
+
+class BaseGatewayLLM(LLM):
+    """Base class for Gateway LLM"""
+    project_id: str
+    model_id: str
+    model_type: str
+    conversation_history: list
+    gateway_auth_token: str
+    alert_log_threshold: float = 0.9
+    logger: logging.Logger | None = None
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.log = LoggerWrapper(self.logger)
```

Comment:
I don't really like it either, but this way you can pass whichever logger you'd like (although it still limits you to the standard `logging` module). I saw that projects use different loggers and I didn't want to assume that it will always be the root one. Plus it helps when it's disabled (not passed as an argument) because you don't have to keep checking if `self.log` exists.